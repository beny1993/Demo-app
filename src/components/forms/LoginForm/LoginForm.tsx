import React from 'react';
import { Form } from 'antd';
import { Input, Button } from '@/components/atoms';

/* eslint-disable no-template-curly-in-string */
const validateMessages = {
  required: '${label} is required!',
};

interface Props {
  onFinish?: (values: any) => Promise<void>;
  loading?: boolean;
}

const LoginForm = ({ onFinish, loading }: Props) => (
  <div className="bg-white rounded-md tracking-wide mb-3 relative">
    <div className="text-xl align-right font-bold mb-5 text-right">ورود</div>
    <Form
      layout="vertical"
      onFinish={onFinish}
      validateMessages={validateMessages}
    >
      <div className="flex flex-col w-full">
        <div className="flex-col">
          <div className="w-full flex flex-col">
            <Form.Item
              label="ایمیل"
              name="username"
              className="flex-1"
              rules={[
                { required: true, message: 'وارد کردن ایمیل ضروری است' },
                {
                  type: 'email',
                  message: 'ایمیل وارد شده معتبر نیست',
                },
              ]}
            >
              <Input placeholder="example@domain.com" className="rounded-md" />
            </Form.Item>
          </div>
        </div>
        <div className="flex flex-col w-full">
          <div className="w-full flex flex-col">
            <Form.Item
              name="password"
              label="رمز عبور"
              className="flex-1"
              rules={[{ required: true, message: 'وارد کردن ایمیل ضروری است' }]}
            >
              <Input placeholder="رمز عبور" type="password" />
            </Form.Item>
          </div>
        </div>
        <div className="flex flex-col w-full">
          <Button
            loading={loading}
            className="text-white h-10 font-bold bg-blue text-md mt-2"
            htmlType="submit"
          >
            ورود
          </Button>
        </div>
      </div>
    </Form>
  </div>
);

export default LoginForm;
