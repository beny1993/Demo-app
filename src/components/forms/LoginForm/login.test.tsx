/* global describe,it */
import { render, fireEvent, screen, act, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import LoginForm from '@/components/forms/LoginForm/LoginForm';
import { initializeStore } from '@/redux/store';

describe('when change password form render', () => {
  let loginButton;
  let component;
  const onFinish = jest.fn();

  beforeEach(async () => {
    const store = initializeStore({});
    component = render(
      <Provider store={store}>
        <LoginForm
          onFinish={onFinish}
          loading={false}
        />
      </Provider>,
    );
    loginButton = await screen.findByRole('button');
  });

  it('inputs render', () => {
    const username = component.container.querySelector('#username');
    const password = component.container.querySelector('#password');
    expect(username).toBeTruthy();
    expect(password).toBeTruthy();
  });

  it('buttons render', () => {
    expect(loginButton).toBeTruthy();
  });

  it('change value works', () => {
    const username = component.container.querySelector('#username');
    const password = component.container.querySelector('#password');
    fireEvent.change(username, { target: { value: 'example@gmail.com' } });
    fireEvent.change(password, { target: { value: '12345678' } });
    expect(username.value).toBe('example@gmail.com');
    expect(password.value).toBe('12345678');
  });

  it('login button works', async () => {
    const username = component.container.querySelector('#username');
    const password = component.container.querySelector('#password');
    fireEvent.change(username, { target: { value: 'm.savalanpour@gmail.com' } });
    fireEvent.change(password, { target: { value: 'm@12345678' } });
    await act(async () => {
      fireEvent.submit(loginButton);
    });
    await waitFor(() => expect(onFinish).toHaveBeenCalledTimes(1));
  });

  it('Login Form Snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
