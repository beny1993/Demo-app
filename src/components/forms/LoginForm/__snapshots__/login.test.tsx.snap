// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when change password form render Login Form Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-white rounded-md tracking-wide mb-3 relative"
      >
        <div
          class="text-xl align-right font-bold mb-5 text-right"
        >
          ورود
        </div>
        <form
          class="ant-form ant-form-vertical"
        >
          <div
            class="flex flex-col w-full"
          >
            <div
              class="flex-col"
            >
              <div
                class="w-full flex flex-col"
              >
                <div
                  class="ant-row ant-form-item flex-1"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class="ant-form-item-required"
                      for="username"
                      title="ایمیل"
                    >
                      ایمیل
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input rounded-md h-12"
                          id="username"
                          placeholder="example@domain.com"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-col w-full"
            >
              <div
                class="w-full flex flex-col"
              >
                <div
                  class="ant-row ant-form-item flex-1"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class="ant-form-item-required"
                      for="password"
                      title="رمز عبور"
                    >
                      رمز عبور
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input undefined h-12"
                          id="password"
                          placeholder="رمز عبور"
                          type="password"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-col w-full"
            >
              <button
                class="ant-btn ant-btn-primary w-full px-4 rounded text-white h-10 font-bold bg-blue text-md mt-2"
                type="submit"
              >
                <span>
                  ورود
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-white rounded-md tracking-wide mb-3 relative"
    >
      <div
        class="text-xl align-right font-bold mb-5 text-right"
      >
        ورود
      </div>
      <form
        class="ant-form ant-form-vertical"
      >
        <div
          class="flex flex-col w-full"
        >
          <div
            class="flex-col"
          >
            <div
              class="w-full flex flex-col"
            >
              <div
                class="ant-row ant-form-item flex-1"
              >
                <div
                  class="ant-col ant-form-item-label"
                >
                  <label
                    class="ant-form-item-required"
                    for="username"
                    title="ایمیل"
                  >
                    ایمیل
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        class="ant-input rounded-md h-12"
                        id="username"
                        placeholder="example@domain.com"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col w-full"
          >
            <div
              class="w-full flex flex-col"
            >
              <div
                class="ant-row ant-form-item flex-1"
              >
                <div
                  class="ant-col ant-form-item-label"
                >
                  <label
                    class="ant-form-item-required"
                    for="password"
                    title="رمز عبور"
                  >
                    رمز عبور
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        class="ant-input undefined h-12"
                        id="password"
                        placeholder="رمز عبور"
                        type="password"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col w-full"
          >
            <button
              class="ant-btn ant-btn-primary w-full px-4 rounded text-white h-10 font-bold bg-blue text-md mt-2"
              type="submit"
            >
              <span>
                ورود
              </span>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
